task/control-plane-agent/src/mgs_sidecar.rs

```diff
@@ -597,14 +668,34 @@ impl SpHandler for MgsHandler {
             (SpComponent::MONORAIL, ComponentAction::Monorail(action)) => {
                 match action {
                     MonorailComponentAction::RequestChallenge => {
+                        use drv_sprot_api::{LifecycleState, SpRot};
+                        let sprot =
+                            SpRot::from(crate::mgs_common::SPROT.get_task_id());
+                        let challenge = match sprot.lifecycle_state() {
+                            Ok(
+                                LifecycleState::Development
+                                | LifecycleState::Unprogrammed
+                                | LifecycleState::EndOfLife,
+                            )
+                            | Err(_) => {
```

Comment:
That was intentional, but I could be persuaded otherwise.

We have checksums / retries in the `SpRot` protocol, so we shouldn't see glitches / spurious failures; if we can't talk to the RoT, something is Very Wrong, and failing into a state where we can debug the system seemed reasonable.

Reply:
The problem is we both want authorized users to be able to debug the system and not open up a back door if something Very Bad has happened with the RoT. otoh, if we can't actually communicate with the RoT at all we're probably not actually going to be able to do anything about that over the tech port.

Can you add a comment that failing open is an intentional choice? My gut is telling me we may need to re-evaluate this in the future.

Reply:
Sure, I added a comment to that effect